### PR TITLE
Redirected the articles about structs to the language reference

### PR DESCRIPTION
**Pull request by pkulikov**
Follow-up of #17120
Contributes to #10726

This PR redirects the following articles:
- [Structs (C# concepts)](https://docs.microsoft.com/en-us/dotnet/csharp/structs)
- [Structs (C# programming guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/structs)
- [Using structs (C# programming guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/using-structs)
- [Structs (Tour of C#)](https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/structs)

### CODE REVIEW

File: docs/csharp/basic-types.md [hunk 1]
```diff
@@ -77,9 +77,10 @@ It is important to understand two fundamental points about the type system in th
...
+- Each type in the CTS is defined as either a *value type* or a *reference type*. This includes all custom types in the .NET class library and also your own user-defined types. Types that you define by using the `struct` or `enum` keyword are value types. For more information about value types, see [Value types](language-reference/builtin-types/value-types.md). Types that you define by using the [class](language-reference/keywords/class.md) keyword are reference types. For more information about reference types, see [Classes](programming-guide/classes-and-structs/classes.md). Reference types and value types have different compile-time rules, and different run-time behavior.
```
Comment (pkulikov) [inline docs/csharp/basic-types.md]: Mentioned the `enum` keyword and referenced the article about value types.

File: docs/csharp/language-reference/compiler-messages/cs0188.md [hunk 2]
```diff
@@ -13,10 +13,8 @@ The 'this' object cannot be used before all of its fields are assigned to
   
  All fields in a `struct` have to be assigned by a constructor before the constructor can call a method in the `struct`.  
   
- If you see this error when trying to initialize a property in a struct constructor, the solution is to change the constructor parameter to specify the backing field instead of the property itself. Auto-implemented properties should be avoided in structs because they have no backing field and therefore cannot be initialized in any way from the constructor.
```
Comment (pkulikov) [inline docs/csharp/language-reference/compiler-messages/cs0188.md]: As far as I understand, beginning with C# 6, auto-implemented properties can be initialized in a struct constructor. So, removing bits about auto-implemented properties from this article.

**Review by pkulikov — COMMENTED** [review]
@BillWagner there are two non-trivial changes (see below). All others are the link updates/removals.

**Review by BillWagner — APPROVED** [review]
Thanks @pkulikov 

This looks great. I'll check the build report to make sure, and then I'll :shipit:

**Pull request merged**